2017/ideas-list-matplotlib.md

```diff
@@ -0,0 +1,189 @@
+# Matplotlib ideas page
+
+These are just starting points.  Your own ideas are most welcome.
+
+Any questions or comments should be asked on
+[matplotlib-devel list](https://mail.python.org/mailman/listinfo/matplotlib-devel).
+
+## Table of contents
+
+* [Layout manager](#layout)
+* [Documentation improvements](#documentation)
+* [International text handling](#text)
+* [Categorical axes](#categorical)
+* [Compound artists](#compound)
+
+---------
+
+<a name="layout"></a>
+## Layout manager
+
+### Abstract
+
+Matplotlib currently has a very simple method to layout the components
+of a figure (axes, titles, tick labels).  This unfortunately makes it
+very common for text to run together or outside of the image.  A
+proper constraint-based layout engine should be able to automatically
+prevent such artists from running together, as well as solve other
+layout difficulties such as twinned axes and colorbars.
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Advanced | Python, Constraint Solving | [@mdboom][] [@tacaswell][] |
+
+### Technical details
+
+This work will likely be based on an existing constraint-solving
+library, such as
+[Cassowary](http://constraints.cs.washington.edu/cassowary/).  Likely
+this would be implemented in two phases: (1) the implementation of
+existing layout abilities in terms of constraints, accessible through
+a new API, and (2) exposing this new implementation through the
+existing APIs.
+
+### Open Source Development Experience
+
+This project involves experience with Python and and understanding of
+linear constraint solvers.  Computer graphics concepts, particularly
+text layout, would also be an asset.
+
+<a name="documentation"></a>
+## Documentation improvements
```

Comment:
This is out of Google of Summer Code scope. Sorry. The students **must** to write code. But this project is suitable for Google Code-in.